src/onemod/pipeline.py

```diff
@@ -281,20 +286,42 @@ def evaluate(
             Path to resources yaml file. Required if `backend` is
             'jobmon'.
 
-        TODO: Add options to run subset of stages
         TODO: Add options to run subset of IDs
 
         """
         if build:
             self.build()
+
+        if stages is not None:
+            for stage_name in stages:
+                if stage_name not in self.stages:
+                    raise ValueError(
+                        f"Stage '{stage_name}' not found in pipeline."
+                    )
+
+            for stage_name in stages:
+                stage: Stage = self.stages.get(stage_name)
+                if not stage.check_required_inputs_exist():
+                    raise ValueError(
+                        f"Required inputs for stage '{stage_name}' are missing. Ensure data dependencies are available before execution."
+                    )
```

Comment:
For the logic here, if we have stage `A`'s inputs depend on stage B's output and the stage `B` is in the subset, we might not require the input from stage `A`. To make the example more concrete, assume we have stage, `A`, `B`, `C` and we have our pipeline as
```
C -> B -> A
```
If we our subset is `{A, B}` we will only require input for `B`. Please let me know if I mis-understood anything

Reply:
You are correct, thanks for catching! I had written a test that should only pass if this was the case but then sabotaged it with a line manually adding the input data prior to `evaluate()`..

Anyway, updated the validation and the test case. There probably is a more efficient/sophisticated way to run this check on upstream dependencies data outputs, but I think it might be a good role for `DataInterface` so haven't spend time on thinking of optimizing the process here. But for now it is a method on `Pipeline`. Let me know if you think this is sufficient for the time being, or if you think it deserves more iteration sooner rather than later.